### PR TITLE
TASK: Create ContentContext with site matching given node

### PR DESCRIPTION
**Pull request by kitsunet**
The `ContentContext` should not be created from the current request
in the `NodeConverter` or `NodeSearchService` but rather be defined
by the node in question.

### CODE REVIEW

**Comment by kitsunet** [comment]
Needs some more test adjustments and some other places should change as well. Will follow up. But good as a first step to see what I mean...

**Comment by kdambekalns** [comment]
@kitsunet any further plans with this one?

**Comment by kitsunet** [comment]
This IMHO needs a serious discussion to go ahead, I think it's the right way, but I don't want to spent agse working on a refactoring of this and then someone comes around and doesn't like the direction...

File: TYPO3.Neos/Classes/TYPO3/Neos/Domain/Service/NodeSearchService.php [hunk 1]
```diff
@@ -62,8 +77,16 @@ public function findByProperties($term, array $searchNodeTypes, Context $context
...
         $nodeDataRecords = $this->nodeDataRepository->findByProperties($term, $nodeTypeFilter, $context->getWorkspace(), $context->getDimensions(), $startingPoint ? $startingPoint->getPath() : null);
+        $baseContextProperties = $context->getProperties();
+        /** @var NodeData $nodeData */
         foreach ($nodeDataRecords as $nodeData) {
-            $node = $this->nodeFactory->createFromNodeData($nodeData, $context);
+            $site = $this->siteRepository->findOneByNodePath($nodeData->getPath());
+            $contextProperties = array_merge($baseContextProperties, [
+                'currentSite' => $site,
+                'currentDomain' => $site->getFirstActiveDomain()
+            ]);
```
Comment (aertmann) [inline TYPO3.Neos/Classes/TYPO3/Neos/Domain/Service/NodeSearchService.php]: this doesn't really belong here, while cleaning up this we should get rid of all those places we have this similar logic..

**Comment by aertmann** [comment]
makes sense to me

**Comment by kitsunet** [comment]
Replaced by https://github.com/neos/neos-development-collection/pull/1572

**Pull request closed without merging**